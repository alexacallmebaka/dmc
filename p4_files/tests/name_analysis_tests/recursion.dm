func : (x : int) int {
  if (x == 0) {
    return x;
  }
  return func(x);
}