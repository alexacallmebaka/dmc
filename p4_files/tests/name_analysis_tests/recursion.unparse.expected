func{(int)->int} : (x{int} : int) int  {
  if (x{int} == 0){
      return x{int};
  }
  return func{(int)->int}(x{int});
}